### PR TITLE
qa/cephfs: allow checking for multiple error messages while...

### PR DESCRIPTION
**Pull request by rishabh-d-dave**
while negative testing.

Related PR - PR https://github.com/ceph/ceph/pull/41779. Commits on this PR originally were located on PR https://github.com/ceph/ceph/pull/41779. These have been moved to a separate PR for ease of review, testing, merging and, more importantly to get these commits merged sooner so that this code can be reused to write other PRs.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
  - [x] improvement in QA
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by rishabh-d-dave** [comment]
jenkins test make check

**Comment by rishabh-d-dave** [comment]
jenkins test make check arm64

**Comment by rishabh-d-dave** [comment]
jenkins test api

**Comment by rishabh-d-dave** [comment]
jenkins test api

**Comment by rishabh-d-dave** [comment]
jenkins test api

**Comment by rishabh-d-dave** [comment]
https://jenkins.ceph.com/job/ceph-api/40158/

File: qa/tasks/cephfs/mount.py [hunk 1]
```diff
@@ -754,44 +754,65 @@ def run_as_root(self, **kwargs):
...
+
+        proc_stderr = proc.stderr.getvalue().lower()
+        msg = ('didn\'t find any of the expected string in stderr.\n'
+               f'expected string: {exp_errmsgs}\n'
+               f'received error message: {proc_stderr}\n'
+               'note: received error message is converted to lowercase')
+        for e in exp_errmsgs:
+            if e in proc_stderr:
+                break
+        else:
```
Comment (vshankar) [inline qa/tasks/cephfs/mount.py:784]: wrong indentation?

Comment (rishabh-d-dave) [reply qa/tasks/cephfs/mount.py:785]: That `else` is meant for `for`.

Comment (rishabh-d-dave) [reply qa/tasks/cephfs/mount.py:785]: Added a comment.

**Review by vshankar — COMMENTED** [review]
Looks fine - please run this through fs suite.

**Review by vshankar — APPROVED** [review]
@rishabh-d-dave please run this through fs suite.

**Comment by rishabh-d-dave** [comment]
It was already a part of last one. The run mostly went fine. Looking forward to discuss the QA run before  posting results.

**Comment by rishabh-d-dave** [comment]
QA was successful.

**Comment by rishabh-d-dave** [comment]
Last PR branch update was to resolve the conflict during rebase. The conflict was between this PR and PR #46991. Here's the change that was made for conflict resolution -
https://github.com/ceph/ceph/compare/d703fb275c3ebfa42eb7063084e6b191b4c6d7f9..82b58e366718d95859ae2e98f24fc2a557efc0e6#diff-9998476d9d02f4136e9f6b4b512be93a77a5911a30027bec6a120706a1a024f3

**Comment by rishabh-d-dave** [comment]
The QA run had both of these PRs as well as changes required for resolving conflict.

```
diff --cc qa/tasks/cephfs/caps_helper.py
index 249cbec361f,ddfff9bc11f..48eae3bdab9
--- a/qa/tasks/cephfs/caps_helper.py
+++ b/qa/tasks/cephfs/caps_helper.py
@@@ -66,8 -67,7 +67,8 @@@ class CapsHelper(CephFSTestCase)
                  if path.find(mount.hostfs_mntpt) != -1:
                      cmdargs.append(path)
                      mount.negtestcmd(args=cmdargs, retval=1,
-                                      errmsg='permission denied')
+                                      errmsgs=possible_errmsgs)
 +                    cmdargs.pop(-1)
```

I'll merge this PR when the CI is green.

**Comment by rishabh-d-dave** [comment]
jenkins test api

**Comment by rishabh-d-dave** [comment]
QA was successful - https://tracker.ceph.com/projects/cephfs/wiki/Main#2022-Jul-15.

**Pull request merged**